pytest/tests/test_migration_service.py

```diff
@@ -0,0 +1,142 @@
+
+        # sanity check that migrating and target node have different keys
+        assert migrating_node.account_id() == target_node.account_id()
+        assert migrating_node.p2p_public_key != target_node.p2p_public_key
+        # sanity check that the migrating node is an active participant in a running state
+        assert running_state_matches_participant_key(
+            cluster, migrating_node.account_id(), migrating_node.p2p_public_key
+        )
+
+        # ensure signature requests are processed
+        cluster.send_and_await_ckd_requests(1)
+        cluster.send_and_await_signature_requests(1)
+
+        # 1. submit backup service info
+        backup_service_info: BackupServiceInfo = backup_service.info()
+        cluster.register_backup_service_info(
+            migrating_node_id, backup_service_info=backup_service_info
+        )
+
+        # wait on the nodes debug endpoint for the backup service to appear
+        expected_migrations.state_by_account[migrating_node.account_id()] = (
+            AccountEntry(backup_service_info, destination_node_info=None)
+        )
+        migrating_node.wait_for_migration_state(expected_migrations)
+
+        # 2. use the backup service cli for a keyshares GET request
+        contract_state = cluster.get_contract_state()
+        backup_service.set_contract_state(contract_state)
+        backup_service.get_keyshares(mpc_node=mpc_nodes[migrating_node_id])
+
+        # 3. node operator initiates node migration
+        signer_account_pk = target_node._signer_key.pk
+        destination_node_participant_info = ParticipantInfo(
+            url=target_node.p2p_url, sign_pk=target_node.p2p_public_key
+        )
+        destination_node = DestinationNodeInfo(
+            signer_account_pk=signer_account_pk,
+            destination_node_info=destination_node_participant_info,
+        )
+        cluster.start_node_migration(migrating_node_id, destination_node)
+
+        # wait for both nodes to register contract change
+        expected_migrations.state_by_account[migrating_node.account_id()] = (
+            AccountEntry(backup_service_info, destination_node)
+        )
+        migrating_node.wait_for_migration_state(expected_migrations)
+        target_node.wait_for_migration_state(expected_migrations)
+
+        # 4. use the backup service cli to PUT keyshares
+        contract_state = cluster.get_contract_state()
+        backup_service.set_contract_state(contract_state)
+        backup_service.put_keyshares(mpc_node=target_node)
+
+        # 5. assert migration was successful
+        n_try = 0
+        while n_try < 50:
+            if running_state_matches_participant_key(
+                cluster, migrating_node.account_id(), target_node.p2p_public_key
+            ):
+                print("successfully migrated node")
+                break
+            else:
+                print("waiting for migration to conclude")
+                time.sleep(1)
+                n_try += 1
```

Comment:
Nit: It would be nice to break this retry logic out to some helper function. It's unfortunate that we don't have a way to await this state change without having to poll.